scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -3116,21 +3115,39 @@ def _pass_thru(
     return res[-len(parent_spec):]
 
 
-def _raw_unit_addr(node: dtlib_Node) -> bytes:
+def _raw_unit_addr(node: dtlib_Node, parent: dtlib_Node) -> bytes:
     # _map_interrupt() helper. Returns the unit address (derived from 'reg' and
     # #address-cells) as a raw 'bytes'
 
+    iparent: Optional[dtlib_Node] = parent
+    addr_len: Optional[int] = None
+
+    while iparent is not None:
+        addr_len = _address_cells_self(iparent)
+        if addr_len is None:
+            # check the parent(s) of interrupt-controller(s) too
+            iparent = iparent.parent
+        else:
+            break
```

Comment:
This seems to treat `#address-cells` as inherited from a node's ancestor, which is contrary to what DTSpec says:
https://github.com/devicetree-org/devicetree-specification/blob/2697c66a59b9177f8ff7813c355cc7aa05f6e40a/source/chapter2-devicetree-basics.rst?plain=1#L644

> The *#address-cells* and *#size-cells* properties are not inherited from ancestors in the devicetree. They shall be explicitly defined.
>
> A DTSpec-compliant boot program shall supply *#address-cells* and *#size-cells* on all nodes that have children.
>
> If missing, a client program should assume a default value of 2 for *#address-cells*, and a value of 1 for *#size-cells*.

Reply:
Thank you for the reference. I'll rework the algo.